### ISSUE TITLE
Docker Extension generates ASP.NET Core Dockerfile with $BUILDPLATFORM Which is not Recognized by Azure Container Registry

### ISSUE THREAD

**Issue by simonkurtz-MSFT** [microsoft/vscode-docker#4149]
Does this issue occur when all extensions are disabled?: No, but it's a Microsoft extension for which I am filing the issue.

- VS Code Version: 1.84.0
- OS Version: Windows 11
- Docker Extension Version: 1.27.0
- .NET SDK Version: 7.0.403
- AZ CLI Version 2.53.1

Steps to Reproduce:

1. From a VS Code terminal, generate a new Web Api: `dotnet new webapi -o Foo`
2. Press CTRL + SHIFT +P, select "Docker: Add Docker Files to Workspace..."
3. Select ".NET: ASP.NET Core"
4. Select "Linux"
5. Select any port.
6. Select "No" for optional Docker Compose files.
7. This generates a _Dockerfile_ with `FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:7.0 AS build`
8. Create an Azure Container Registry and attempt a build:

```powershell
$RANDOM_STRING=-join ((97..122) + (48..57) | Get-Random -Count 6 | ForEach-Object { [char]$_})
$RESOURCE_GROUP="rg-$RANDOM_STRING"
$LOCATION="eastus"
$AZURE_CONTAINER_REGISTRY_NAME="cr$RANDOM_STRING"

az group create `
--name $RESOURCE_GROUP `
--location "$LOCATION"

az acr create `
--resource-group $RESOURCE_GROUP `
--name $AZURE_CONTAINER_REGISTRY_NAME `
--sku Basic `
--admin-enabled true

az acr build `
--registry $AZURE_CONTAINER_REGISTRY_NAME `
--image "foo" `
--file 'Foo/Dockerfile' .
```

Observe the following error:

```
unable to understand line FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:7.0 AS build
failed to run step ID: build: failed to scan dependencies: exit status 1

Run ID: ca1 failed after 3s. Error: failed during run, err: exit status 1
Run failed
```

Given that the ASP.NET Core image is a multi-platform image, the build image makes sense as it tailors to the architecture upon which it is executing the build. What I can't quite make sense of is how to best marry this with a build that is done in Azure Container Registry. This may well not be something that the extension can address, but I am filing it here as a first start since that's where it originates. What should be the appropriate path to generate a _Dockerfile_ with the extension and subsequently run it in ACR?

## Mitigations
- Remove `--platform=$BUILDPLATFORM` from the _Dockerfile_. This means Azure Container Registry won't need to know about it. I suspect that the internal build that occurs via the Azure Container Registry takes care of this differently to optimize for its platform.

**Comment by bamurtaugh** [comment]
Transferred! And assigning Christof and myself (as we work on the Dev Containers extension).

**Comment by bwateratmsft** [comment]
BUILDPLATFORM is set by Docker during build so it might be worth having asking ACR to do the same.

**Comment by simonkurtz-MSFT** [comment]
@bwateratmsft, makes sense to me. Is that something you all can facilitate with the ACR team by way of an issue? Might be more impactful coming from the PG.

**Comment by bwateratmsft** [comment]
Looks like this is already known by the ACR team; https://github.com/Azure/acr/issues/697 and https://github.com/Azure/acr-builder/issues/449. We'll see if we can draw more attention to it.

**Comment by simonkurtz-MSFT** [comment]
@bwateratmsft, thank you! Our workaround at this time is to simply remove `--platform=$BUILDPLATFORM`.

**Comment by simonkurtz-MSFT** [comment]
[like]  Simon Kurtz reacted to your message:
________________________________
